packages/astro/src/runtime/server/render/page.ts

```diff
@@ -157,6 +157,11 @@ export async function renderPage(
 										}
 									}
 								}
+								if (chunk instanceof Response) {
+									throw new AstroError({
+										...AstroErrorData.ResponseSentError,
+									});
+								}
```

Comment:
Because we now eagerly render components, even redirects in nested components can now render successfully, as the response hasn't started sending at that point. So a response returned from a nested component, which will cause the following function to crash. So I'm catching this here again, but I'm not sure if this is the right place or the right way.

Reply:
I believe it's really valuable to add your comment **in the code**, so we don't lose this piece of knowledge!